cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/store/AppMetadataStoreTest.java

```diff
@@ -191,13 +188,13 @@ public void testInvalidStatusPersistence() throws Exception {
 
   private void assertPersistedStatus(final AppMetadataStore metadataStoreDataset, TransactionExecutor txnl,
                                      final long startSourceId, final long runningSourceId,
-                                     final long killedSourceId, ProgramRunStatus expectedRunStatus) throws Exception {
+                                     final long killedSourceId, final ProgramRunStatus expectedRunStatus)
+    throws Exception {
     // Add some run records
     ApplicationId application = NamespaceId.DEFAULT.app("app");
     final ProgramId program = application.program(ProgramType.WORKFLOW, "program");
     // use sourceId's to ensure runId is unique for each run
-    final RunId runId = RunIds.generate(10000 + startSourceId + runningSourceId + killedSourceId);
-    final AtomicReference<RunRecordMeta> resultRecord = new AtomicReference<>();
+    final RunId runId = RunIds.generate(startSourceId * 3 + runningSourceId * 5 + killedSourceId * 7);
```

Comment:
it's still possible, though unlikely, that you can get the same runid. Just pass in the value.